R/filterHaplotype.R

```diff
@@ -117,56 +118,52 @@ createHaplotypOverviewTable <- function(allHaplotypesFilenames, clusterFilenames
   
   ######
   ## singleton $ representativ
-  env <- environment()
   overviewHap$singelton <- T
   overviewHap$representatives <- F
-  env$overviewHap$singelton <- overviewHap$singelton
-  env$overviewHap$representatives <- overviewHap$representatives
   
   repfile <- clusterFilenames[,"RepresentativeFile"]
-  haplotypes <- lapply(seq_along(repfile), function(i){
-  	sr1 <- readFasta(repfile[i])
-  	vec <- do.call(rbind, strsplit(as.character(id(sr1)), "_"))
-  	clusterResp <- vec[,1]
-  	clusterSize <- as.integer(vec[,2])
-  	env$overviewHap[clusterResp,"representatives"] <- T
-  	clusterResp <- clusterResp[clusterSize>1]
-  	env$overviewHap[clusterResp,"singelton"] <- F
-  	return(NULL)
-  })
-  
-  table(env$overviewHap$singelton)
-  table(env$overviewHap$representatives)
-  overviewHap$singelton <- env$overviewHap$singelton
-  overviewHap$representatives <- env$overviewHap$representatives
-  
-  
-  ######
-  # INDEL - Homopolymere
-	if(!is.null(referenceSequence)){
-		idx <- as.character(id(sr1)) %in% as.character(overviewHap$HaplotypesName[!overviewHap$singelton])
-		sr1 <- sr1[idx]
-		aln1 <- pairwiseAlignment(sread(sr1), referenceSequence, type="global")
-		idx <- (lengths(deletion(aln1)) + lengths(insertion(aln1))) == 0
-		names(idx) <- as.character(id(sr1))
-		overviewHap[names(idx),"indels"] <- !idx
-		sr1 <- sr1[idx]
-	}
-  
-  ###
-  # only SNP variation - Final Haplotypes
-  if(!is.null(snpList)){
-	  snps <- lapply(as.integer(snpList[,"Pos"]), function(n){
-	    as.character(subseq(sread(sr1), start=n, width = 1))
-	  })
-	  # snps <- do.call(paste0, snps)
-	  snps <- apply(do.call(cbind, snps), 1, paste, collapse="")
-	  names(snps) <- as.character(id(sr1))
-	  
-	  overviewHap[names(snps),"snps"] <- snps
-  }else{
-  	overviewHap[,"snps"] <- ""
+  for (fn in repfile){
+    sr1 <- readFasta(fn)
+    vec <- do.call(rbind, strsplit(as.character(id(sr1)), "_"))
+    at <- match(vec[,1], overviewHap$HaplotypesName)
+    clusterSize <- as.integer(vec[,2])
+    overviewHap$representatives[at] <- T
+    overviewHap$singelton[at[clusterSize>1]] <- F
   }
   
+  #  Indels
+  overviewHap$indels <- F
+  hap_set <- as.character(with(overviewHap, HaplotypesName[representatives & ! singelton]))
+	haps <- readDNAStringSet(allHaplotypesFilenames)[hap_set]
```

Comment:
Why do you read the allHaplotypesFilenames and not reuse the representative sequence reads? I also think that it disables the possibility of my version for length-polymorphic handling.
I'm happy to include your handling in future, but maybe as a different function. Let us discuss this after the workshop.